shortcodes/class.geojson-shortcode.php

```diff
@@ -106,21 +114,33 @@ function layerStyle (feature) {
                 }      
                 function onEachFeature (feature, layer) {
                     var props = feature.properties || {},
-                        text = popup_property && props[ popup_property ] || template(popup_text, feature.properties);
+                        text = popup_property && props[ popup_property ] || template(popup_text, feature.properties),
+                        featureName = feature.properties.name,
+                        iconUrl = feature.properties.iconUrl;
                     if (text) {
                         layer.bindPopup( text );
                     }
+                    if (iconUrl) {
+                        // create the legend entities for the feature names, assumes the filename pattern is *-[iconColor].*
+                        legendVals[featureName] = iconUrl.slice(iconUrl.lastIndexOf("-")+1, iconUrl.lastIndexOf("."));
+                    }
                 }
-                var templateRe = /\{ *([\w_-]+) *\}/g;
-                function template(str, data) {
-                    return str.replace(templateRe, function (match, key) {
-                        var value = data[key];
-                        if (value === undefined) {
-                            return match;
-                        }
-                        return value;
-                    });
-                }  
+                layer.on('ready', function () {
+                    if('<?php echo $legend; ?>' && JSON.stringify(legendVals) !== JSON.stringify({})) {
+                        // add a legend for the icons
+                        var legend = L.control({position: 'bottomright'});
+                        legend.onAdd = function (previous_map) {
```

Comment:
camel case for javascript: `previousMap`

Reply:
Good eye. I guess I was following the original repo coding style where I shouldn't have.